cdap-ui/app/cdap/components/PostRunActions/PostRunActionsWizard.js

```diff
@@ -0,0 +1,281 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import IconSVG from 'components/IconSVG';
+import {UncontrolledTooltip} from 'components/UncontrolledComponents';
+import {getCurrentNamespace} from 'services/NamespaceStore';
+import AvailablePluginsStore from 'services/AvailablePluginsStore';
+import {MyArtifactApi} from 'api/artifact';
+import {generateNodeConfig} from 'services/HydratorPluginConfigFactory';
+import { ModalBody } from 'reactstrap';
+import HydratorModal from 'components/HydratorModal';
+import SelectWithOptions from 'components/SelectWithOptions';
+import KeyValuePairs from 'components/KeyValuePairs';
+import RadioGroup from 'components/RadioGroup';
+import DSVEditor from 'components/DSVEditor';
+import {convertMapToKeyValuePairsObj} from 'components/KeyValuePairs/KeyValueStoreActions';
+import uuidV4 from 'uuid/v4';
+import {preventPropagation} from 'services/helpers';
+
+export default class PostRunActionsWizard extends Component {
+  static propTypes = {
+    action: PropTypes.object,
+    isOpen: PropTypes.bool,
+    toggleModal: PropTypes.func
+  };
+
+  state = {
+    groupsConfig: {}
+  };
+
+  componentWillMount() {
+    if (!Object.keys(this.props.action._backendProperties || {}).length) {
+      this.pluginFetch(this.props.action);
+    } else {
+      this.fetchWidgets(this.props.action);
+    }
+  }
+
+  // Fetching Backend Properties
+  pluginFetch(action) {
+    let {name, version, scope} = action.plugin.artifact;
+    let params = {
+      namespace: getCurrentNamespace(),
+      artifactId: name,
+      version,
+      scope,
+      extensionType: action.plugin.type,
+      pluginName: action.plugin.name
+    };
+
+    MyArtifactApi.fetchPluginDetails(params)
+      .subscribe(res => {
+        this.props.action._backendProperties = res[0].properties;
+        this.fetchWidgets(action);
+      });
+  }
+
+  // Fetching Widget JSON for the plugin
+  fetchWidgets(action) {
+    let pluginsMap = AvailablePluginsStore.getState().plugins.pluginsMap;
+    let {name, version, scope} = action.plugin.artifact;
+    let actionNameType = ''.concat(action.plugin.name, '-').concat(action.plugin.type);
+    let postRunActionPluginKey = ''.concat(actionNameType, '-').concat(name, '-').concat(version, '-').concat(scope);
+    let pluginJson = pluginsMap[postRunActionPluginKey];
+    let groupsConfig = generateNodeConfig(this.props.action._backendProperties, pluginJson.widgets);
+    this.setState({
+      groupsConfig
+    });
+  }
+
+  getComponentFromWidgetType(type, name, attributes) {
+    let Component;
+    let props;
+    let value = this.props.action.plugin.properties[name];
+    let onChangeFn = () => {};
```

Comment:
it's not doing anything?

Reply:
Since we only care about PostRunActionsWizard in detail view (for now), everything is disabled, so we don't need to handle any changes. However I'm still passing a function for `DSVEditor` and `KeyValuePairs`, since those two expect `onChange` as a required property.

Reply:
Should we perhaps modify the DSVEditor and KeyValuePairs to make the onChange as optional? seems odd to have to pass a function if it's not being used..